src/helpers/facetHelper.ts

```diff
@@ -0,0 +1,161 @@
+
+type allFacetResultTypes = (ProductAssortmentFacetResult | ContentAssortmentFacetResult | ProductCategoryAssortmentFacetResult | BrandFacetResult | CategoryFacetResult | CategoryHierarchyFacetResult | ContentDataObjectFacetResult | ContentDataDoubleRangeFacetResult | ContentDataDoubleRangesFacetResult | ContentDataStringValueFacetResult | ContentDataBooleanValueFacetResult | ContentDataDoubleValueFacetResult | ContentDataIntegerValueFacetResult | DataObjectFacetResult | DataObjectDoubleRangeFacetResult | DataObjectDoubleRangesFacetResult | DataObjectStringValueFacetResult | DataObjectBooleanValueFacetResult | DataObjectDoubleValueFacetResult | PriceRangeFacetResult | PriceRangesFacetResult | ProductCategoryDataObjectFacetResult | ProductCategoryDataDoubleRangeFacetResult | ProductCategoryDataDoubleRangesFacetResult | ProductCategoryDataStringValueFacetResult | ProductCategoryDataBooleanValueFacetResult | ProductCategoryDataDoubleValueFacetResult | ProductDataObjectFacetResult | ProductDataDoubleRangeFacetResult | ProductDataDoubleRangesFacetResult | ProductDataStringValueFacetResult | ProductDataBooleanValueFacetResult | ProductDataDoubleValueFacetResult | ProductDataIntegerValueFacetResult | RecentlyPurchasedFacetResult | VariantSpecificationFacetResult);
+
+export function getFacets(
+    context: FacetContext,
+    facetBuilder: FacetBuilder,
+    filters: Record<string, string | string[]>, 
+    facets: ProductFacetResult | null | undefined) {
+
+    const facetsToAdd = facetConfig.filter(x => x.contexts.includes(context));
+
+    facetsToAdd.forEach(facetToAdd => {
+        switch(facetToAdd.type) {
+        case 'BrandFacet':
+            addBrandFacet(facetBuilder, filters);
+            break;
+        case 'Category':
+            addCategoryFacet(facetBuilder, filters);
+            break;
+        case 'DataString':
+            addDataStringFacet(facetToAdd.dataKey, facetToAdd.dataSelectionStrategy, facetBuilder, filters);
+            break;
+        case 'SalesPrice':
+            addSalesPriceFacet(facetBuilder, filters, facets);
+            break;
+        case 'DataDouble':
+            addDataDoubleFacet(facetToAdd.dataKey, facetToAdd.dataSelectionStrategy, facetBuilder, filters);
+            break;
+        case 'DataDoubleRange':
+            addDataDoubleRangeFacet(
+                facetToAdd.dataKey, 
+                facetToAdd.dataSelectionStrategy,
+                facetBuilder,
+                filters,
+                facets);
+            break;
+        default:
+            console.error(`Could not handle facet configuration with type '${facetToAdd.type}'`);
+        }
+    });
+}
+
+export function getFacetConfigEntryForResult(facetResult: allFacetResultTypes) {
```

Comment:
This name is confusing me a bit. What does "ForResult" mean here?
Could it not just be:
```suggestion
export function getFacetSettings(facet: allFacetResultTypes) {
```

Reply:
Check!